cumulusci/utils/yaml/tests/test_cumulusci_yml.py

```diff
@@ -37,3 +42,39 @@ def test_converter_is_selective(self):
 
         rc = _replace_nbsp(inp)
         assert rc == inp
+
+    @patch("cumulusci.utils.yaml.cumulusci_yml.yaml.safe_load")
+    def test_scanner_error(self, safe_load):
+        problem_mark = Mock(line=12345, column=54321)
+        safe_load.side_effect = ScannerError(problem_mark=problem_mark)
+
+        f_config = Mock()
+        f_config.read.return_value = "xyz: abc \n >+>:>%*"  # return invalid yaml
+        # mock f_config.name
+        type(f_config).name = PropertyMock(return_value="cumulusci.yml")
+
+        with pytest.raises(CumulusCIException) as error:
+            cci_safe_load(f_config)
+
+        assert error.typename == "CumulusCIException"
+        assert (
+            "An error occurred parsing cumulusci.yml at line 12345, column 54321.\nError message: None"
+            == error.value.args[0]
+        )
```

Comment:
Could you rewrite this test to use an actual poorly formatted yaml file in a tempdir instead of mocks? The mocks make it hard to know if we're setting up the mocks realistically.

Also do a test to make sure cci_safe_load still works if we pass in a StringIO() instance instead of a file, because I think they probably don't have a `.name`